### PR TITLE
Contributing updates [triage-skip]

### PR DESCRIPTION
**Pull request by jshcrowthe**
Update the CONTRIBUTING.md to better communicate community expectations.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,39 +1,123 @@
...
+
+## <a name="coc"></a> Code of Conduct
+
+As contributors and maintainers of the Firebase JS SDK project, we pledge to respect everyone who contributes by posting issues, updating documentation, submitting pull requests, providing feedback in comments, and any other activities.
+
+Communication through any of Firebase's channels (GitHub, StackOverflow, Google+, Twitter, etc.) must be constructive and never resort to personal attacks, trolling, public or private harassment, insults, or other unprofessional conduct.
+
+We promise to extend courtesy and respect to everyone involved in this project regardless of gender, gender identity, sexual orientation, disability, age, race, ethnicity, religion, or level of experience. We expect anyone contributing to the project to do the same.
+
+If any member of the community violates this code of conduct, the maintainers of the Firebase JS SDK project may take action, removing issues, comments, and PRs or blocking accounts as deemed appropriate.
```
Comment (paulb777) [inline CONTRIBUTING.md]: See https://groups.google.com/a/google.com/forum/#!msg/opensource-discuss/aBFlzXw_kBw/KrZuKGh0AwAJ

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,39 +1,123 @@
...
+
+If you are subject to or witness unacceptable behavior, or have any other concerns, please drop us a line at samstern@google.com.
```
Comment (paulb777) [inline CONTRIBUTING.md]: An alias to a group might be better - firebase-code-of-conduct@google.com

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,39 +1,123 @@
...
+We promise to extend courtesy and respect to everyone involved in this project regardless of gender, gender identity, sexual orientation, disability, age, race, ethnicity, religion, or level of experience. We expect anyone contributing to the project to do the same.
+
+If any member of the community violates this code of conduct, the maintainers of the Firebase JS SDK project may take action, removing issues, comments, and PRs or blocking accounts as deemed appropriate.
+
+If you are subject to or witness unacceptable behavior, or have any other concerns, please drop us a line at samstern@google.com. 
+
+
+## <a name="question"></a> Got a Question?
+
+If you have questions about how to use the Firebase JS SDK, please direct these to [StackOverflow][stackoverflow] and use the `firebase` tag. You can also use the [Firebase Google Group][firebase-google-group] or [Slack][slack] to contact members of the Firebase team for help.
```
Comment (paulb777) [inline CONTRIBUTING.md]: and` javascript` tag

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,39 +1,123 @@
...
+
+## <a name="other-issue"></a> Production Issues
+
+If you have a production issue, please [contact Firebase support][support] who will work with you to resolve the issue. 
+
+## <a name="submit"></a> Submission Guidelines
+
+### Submitting an Issue
+
+Before you submit your issue search [past issues][archive], [StackOverflow][stackoverflow], and the [Firebase Google Group][firebase-google-group] maybe your question was already answered.
```
Comment (paulb777) [inline CONTRIBUTING.md]: Comma after issue

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,39 +1,123 @@
...
+
+If your issue appears to be a bug, and hasn't been reported, open a new issue. Help us to maximize the effort we can spend fixing issues and adding new features, by not reporting duplicate issues. Providing the following information will increase the chances of your issue being dealt with quickly:
```
Comment (paulb777) [inline CONTRIBUTING.md]: Delete "Help ..." sentence. It was just pointed out in the previous sentence - "Before ...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,39 +1,123 @@
...
-### The small print
+  ```shell
+  git checkout -b my-fix-branch master
+  ```
 
-Contributions made by corporations are covered by a different agreement than the
-one above, the [Software Grant and Corporate Contributor License
-Agreement](https://cla.developers.google.com/about/google-corporate).
+* Create your patch, **including appropriate test cases**. Patches with tests are more likely to be merged.
+* Avoid checking in files that shouldn't be tracked (e.g `node_modules`, `gulp-cache`, `.tmp`, `.idea`). We recommend using a [global](#global-gitignore) gitignore for this.
```
Comment (paulb777) [inline CONTRIBUTING.md]: Don't we have a .gitignore in the project already? The #global-gitignore link goes nowhere.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,39 +1,123 @@
...
+That's it! Thank you for your contribution!
+
+## <a name="commit-message-guidelines"></a> Commit Message Guidelines
 
 This repository follows the commit message format defined by the 
 [validate-commit-msg](https://npm.im/validate-commit-msg) package on NPM. This is
 to make the git history easy to follow, and make it easier to identify which
 commits are associated with features, bugfixes, etc.
 
 We are also Commitizen friendly! If you have the [Commitizen CLI](https://npm.im/commitizen) installed
```
Comment (paulb777) [inline CONTRIBUTING.md]: Comma after CLI

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,39 +1,123 @@
...
-Before you start working on a larger contribution, you should get in touch with
-us first through the issue tracker with your idea so that we can help out and
-possibly guide you. Coordinating up front makes it much easier to avoid
-frustration later on.
+Before we can use your code, you must sign the [Google Individual Contributor License Agreement][google-cla] (CLA), which you can do online. The CLA is necessary mainly because you own the copyright to your changes, even after your contribution becomes part of our codebase, so we need your permission to use and distribute your code. We also need to be sure of various other things—for instance that you'll tell us if you know that your code infringes on other people's patents. You don't have to sign the CLA until after you've submitted your code for review and a member has approved it, but you must do it before we can put your code into our codebase. Before you start working on a larger contribution, you should get in touch with us first through the issue tracker with your idea so that we can help out and possibly guide you. Coordinating up front makes it much easier to avoid frustration later on.
```
Comment (samtstern) [inline CONTRIBUTING.md]: nit: 'things—for' is strange punctuation.

Comment (samtstern) [inline CONTRIBUTING.md]: Also I'd split this paragraph into two.  One about the CLA, one about large PRs.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,39 +1,123 @@
...
+* Test your changes locally to ensure everything is in good working order:
+
+    ```shell
+   npm test
+    ```
+
+* Push your branch to GitHub:
+
+    ```shell
+    git push origin my-fix-branch
```
Comment (samtstern) [inline CONTRIBUTING.md]: If they're following along they have not forked yet, so at this point `origin` is our repo and this command will fail.

Comment (jshcrowthe) [inline CONTRIBUTING.md]: Great point. I'll put in a link to fork the repo first.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: This is an adaptation of the functions-samples CoC. If we later want to update this that shouldn't be too bad.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Great idea! I'll route this there instead.

File: CONTRIBUTING.md (hunk 3 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 4 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 5 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Moved the sentence up as explaining the rationale seems like a good idea to me.

File: CONTRIBUTING.md (hunk 6 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Yup, I rehashed this sentence.

File: CONTRIBUTING.md (hunk 8 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 7 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Can't see exactly what you're referring to, the phrase "CLI" only occurs once in the whole doc.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,39 +1,126 @@
...
+
+## <a name="other-issue"></a> Production Issues
+
+If you have a production issue, please [contact Firebase support][support] who will work with you to resolve the issue. 
+
+## <a name="submit"></a> Submission Guidelines
+
+### Submitting an Issue
+
+Before you submit your issue, search [past issues][archive], [StackOverflow][stackoverflow], and the [Firebase Google Group][firebase-google-group] maybe your question was already answered. You can help us to maximize the effort we spend fixing issues, and adding new features, by not reporting duplicate issues.
```
Comment (hiranya911) [inline CONTRIBUTING.md]: Revise wording in this sentence.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,39 +1,126 @@
...
-know that your code infringes on other people's patents. You don't have to sign
-the CLA until after you've submitted your code for review and a member has
-approved it, but you must do it before we can put your code into our codebase.
-Before you start working on a larger contribution, you should get in touch with
-us first through the issue tracker with your idea so that we can help out and
-possibly guide you. Coordinating up front makes it much easier to avoid
-frustration later on.
+Before we can use your code, you must sign the [Google Individual Contributor License Agreement][google-cla] (CLA), which you can do online. The CLA is necessary mainly because you own the copyright to your changes, even after your contribution becomes part of our codebase, so we need your permission to use and distribute your code. We also need to be sure of various other things, for instance, that you'll tell us if you know that your code infringes on other people's patents. You don't have to sign the CLA until after you've submitted your code for review and a member has approved it, but you must do it before we can put your code into our codebase. 
+
```
Comment (hiranya911) [inline CONTRIBUTING.md]: Mention that a bot will guide them through this process, when they submit a PR.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,39 +1,126 @@
...
+Before you start working on a larger contribution, you should get in touch with us first through the issue tracker with your idea so that we can help out and possibly guide you. Coordinating up front makes it much easier to avoid frustration later on.
```
Comment (hiranya911) [inline CONTRIBUTING.md]: Perhaps mention that big changes may also require other internal review and approvals (launchcals, API reviews etc), which may take additional time.

File: CONTRIBUTING.md (hunk 10 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 11 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 12 above)
Comment (jshcrowthe) [inline CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 7 above)
Comment (paulb777) [inline CONTRIBUTING.md]: oops, after installed - If you have the Commitizen CLI installed,

**Comment by paulb777** [comment]
Should the Code of Conduct be a separate document? From https://github.com/firebase/firebase-ios-sdk/community, it's what's missing to get to 100% on recommended community standards.

**Comment by jshcrowthe** [comment]
I think we could cross post it there sure! It's a good thing for people to read.

**Comment by hiranya911** [comment]
LGTM

**Pull request merged**